utils/python/CIME/system_test.py

```diff
@@ -361,15 +361,104 @@ def _create_newcase_phase(self, test):
     def _xml_phase(self, test):
     ###########################################################################
 
-        test_case = CIME.utils.parse_test_name(test)[0]
-        envtest = EnvTest(self._get_test_dir(test))
+        # modify the CASEROOT xml files depending on the settings of case_opts from
+        # then call case.flush()
 
+        # Determine list of component classes that this coupler/driver knows how
+        # to deal with. This list follows the same order as compset longnames follow.
         files = Files()
         drv_config_file = files.get_value("CONFIG_DRV_FILE")
-        logger.debug("Found drv_config_file %s" % drv_config_file)
-
         drv_comp = Component(drv_config_file)
+        component_classes = drv_comp.get_valid_model_components()
+
+        # Set up a case object
+        case = Case(self._get_test_dir(test)) 
+
+        # Determine the test_case from the test name
+        test_case = CIME.utils.parse_test_name(test)[0]
+        case_opts = CIME.utils.parse_test_name(test)[1] 
+
+        # Determine case_opts from the test_case
+        if case_opts is not None:
+            logger.info("case_opts are %s " %case_opts) 
+            for opt in case_opts:
+
+                logger.debug("case_opt is %s" %opt)
+                if opt == 'D':
+                    case.set_value("DEBUG", True)
+                    logger.info (" DEBUG set to TRUE")
+
+                if opt == 'E':
+                    case.set_value("USE_ESMF_LIB", True)
+                    case.set_value("COMP_INTERFACE", "ESMF")
+                    logger.info (" USE_ESMF_LIB set to TRUE")
+                    logger.info (" COMP_INTERFACE set to ESMF")
+
+                if opt == 'CG':
+                    case.set_value("CALENDAR", "GREGORIAN")
+                    logger.info (" CALENDAR set to %s" %opt)
+
```

Comment:
All your re.searches start with a beginning of line symbol ('^'). You could replace all of these with:

    re.match(r'', opt)

... and you could get rid of the ('^') symbol.